src/storage/btree.c

```diff
@@ -8612,27 +8600,27 @@ btree_get_subtree_capacity (THREAD_ENTRY * thread_p, BTID_INT * btid, PAGE_PTR p
 	  cpc->tot_free_space += cpc2.tot_free_space;
 	  cpc->tot_space += cpc2.tot_space;
 	  cpc->tot_used_space += cpc2.tot_used_space;
-	  pgbuf_unfix_and_init (thread_p, page);
+
+	  cpc->ovfl_oid_pg.tot_free_space += cpc2.ovfl_oid_pg.tot_free_space;
+	  cpc->ovfl_oid_pg.tot_pg_cnt += cpc2.ovfl_oid_pg.tot_pg_cnt;
+	  cpc->ovfl_oid_pg.tot_space += cpc2.ovfl_oid_pg.tot_space;
+	  cpc->ovfl_oid_pg.dis_key_cnt += cpc2.ovfl_oid_pg.dis_key_cnt;
+	  cpc->ovfl_oid_pg.tot_val_cnt += cpc2.ovfl_oid_pg.tot_val_cnt;
+	  if (cpc->ovfl_oid_pg.max_pg_cnt_per_key < cpc2.ovfl_oid_pg.max_pg_cnt_per_key)
+	    {
+	      cpc->ovfl_oid_pg.max_pg_cnt_per_key = cpc2.ovfl_oid_pg.max_pg_cnt_per_key;
+	    }
 	}			/* for */
-      cpc->avg_val_per_key = ((cpc->dis_key_cnt > 0) ? (cpc->tot_val_cnt / cpc->dis_key_cnt) : 0);
-      cpc->nleaf_pg_cnt += 1;
-      cpc->tot_pg_cnt += 1;
-      cpc->tot_free_space += free_space;
-      cpc->tot_space += DB_PAGESIZE;
-      cpc->tot_used_space += (DB_PAGESIZE - free_space);
-      cpc->avg_key_len = ((cpc->dis_key_cnt > 0) ? ((int) (cpc->sum_key_len / cpc->dis_key_cnt)) : 0);
-      cpc->avg_rec_len = ((cpc->dis_key_cnt > 0) ? ((int) (cpc->sum_rec_len / cpc->dis_key_cnt)) : 0);
-      cpc->avg_pg_key_cnt = ((cpc->leaf_pg_cnt > 0) ? ((int) (cpc->dis_key_cnt / cpc->leaf_pg_cnt)) : 0);
-      cpc->avg_pg_free_sp = ((cpc->tot_pg_cnt > 0) ? (cpc->tot_free_space / cpc->tot_pg_cnt) : 0);
     }
   else
     {				/* a leaf page */
+      int free_space_ovfl = 0;
```

Comment:
The below definition of variables looks better to move down into "if clause"
```
      int free_space_ovfl = 0;
      int oid_cnt_ovfl = 0;
      int pg_cnt_per_key = 0;
```

Reply:
I will fix it.  thank you.